syntax/nickel.vim

```diff
@@ -73,9 +73,11 @@ syntax region nickelEnumTag start=+'"+ skip=+\\"+ end=+"+ oneline
 " Enum type
 syntax region nickelTypeContract start=+\[|+ end=+|]+ contains=nickelEnumTag
 
+syntax match nickelInterpolation "\v\%\{([^\}])*\}"
```

Comment:
I don't think this definition with only a regex will work well, because the interpolated expression can be an arbitrary value, including something with `{`/`}`:

```nickel
"Hello, %{ {value = "world"}.value }"
```

I think you need to use a region, which can contain an arbitrary Nickel expression, as is done e.g. in the Nix vim plugin: https://github.com/LnL7/vim-nix/blob/7d23e97d13c40fcc6d603b291fe9b6e5f92516ee/syntax/nix.vim#L35

Reply:
This is a bit too complicated for now. In most cases it works. I would suggest to make a new issue for full syntax interpolation and for now accept this half finished PR.

Reply:
Fair enough.